.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,17 @@
+name: Build and Test
+on: [push]
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "This job was triggered by a ${{ github.event_name }} event."
+      - run: echo "Running on OS ${{ runner.os }}"
+      - run: echo "Checking out branch ${{ github.ref }}."
+      - name: Check out repository code
+        uses: actions/checkout@v3
```

Comment:
nit: in a follow-up I can pin actions by hashes & use depandabot to update them

Reply:
just did it, using pin-github-action https://github.com/mheap/pin-github-action, manually confirming it's the right hash https://github.com/actions/checkout/commit/8e5e7e5ab8b370d6c329ec480221332ada57f0ab